core/src/main/java/hudson/model/Job.java

```diff
@@ -1143,6 +1148,22 @@ private HealthReport getBuildStabilityHealthReport() {
         int failCount = 0;
         int totalCount = 0;
         RunT i = getLastBuild();
+        RunT u = getLastFailedBuild();
+        if (i != null && u == null) {
+                // no failures, like ever
```

Comment:
I would be in favor of that. I am not sure `RunMap.getLoadedBuilds` as currently designed suffices, because it does not tell you which unmentioned build numbers in fact exist on disk—without actually trying to load them, you cannot tell whether they are there but unloaded, or have been deleted. There are a variety of reasons I think exposing `numberOnDisk` as an API in some form would be useful; this is one, as is a better fix of https://github.com/jenkinsci/jenkins/commit/0449883d152a772deea8621ce2e2c3e97e7e8a1f (JENKINS-23945).